code/modules/vehicles/armored/interiors/chairs.dm

```diff
@@ -6,119 +6,84 @@
 	resistance_flags = RESIST_ALL
 	dir = EAST
 
-
-/obj/structure/bed/chair/vehicle_driver_seat
+/obj/structure/bed/chair/vehicle_crew
 	name = "driver seat"
 	icon = 'icons/obj/armored/3x3/tank_interior.dmi'
 	icon_state = "vehicle_chair"
 	resistance_flags = RESIST_ALL
 	dir = EAST
 	///owner of this object, assigned during interior linkage
 	var/obj/vehicle/sealed/armored/owner
+	///The skill required to man this chair
+	var/skill_req = SKILL_LARGE_VEHICLE_VETERAN
+	///modifier to view range when in this chair. Not additive mod
+	var/vis_range_mod
 
-/obj/structure/bed/chair/vehicle_driver_seat/Destroy()
+/obj/structure/bed/chair/vehicle_crew/Destroy()
 	owner = null
 	return ..()
 
-/obj/structure/bed/chair/vehicle_driver_seat/link_interior(datum/interior/link)
+/obj/structure/bed/chair/vehicle_crew/link_interior(datum/interior/link)
 	if(!istype(link, /datum/interior/armored))
 		CRASH("invalid interior [link.type] passed to [name]")
 	var/datum/interior/armored/inside = link
 	owner = inside.container
+	vis_range_mod = owner.vis_range_mod
 
-/obj/structure/bed/chair/vehicle_driver_seat/buckle_mob(mob/living/buckling_mob, force, check_loc, lying_buckle, hands_needed, target_hands_needed, silent)
-	if(buckling_mob.skills.getRating(SKILL_LARGE_VEHICLE) < SKILL_LARGE_VEHICLE_VETERAN)
+/obj/structure/bed/chair/vehicle_crew/buckle_mob(mob/living/buckling_mob, force, check_loc, lying_buckle, hands_needed, target_hands_needed, silent)
+	if(buckling_mob.skills.getRating(SKILL_LARGE_VEHICLE) < skill_req)
 		return FALSE
 	return ..()
 
-/obj/structure/bed/chair/vehicle_driver_seat/post_buckle_mob(mob/buckling_mob)
+/obj/structure/bed/chair/vehicle_crew/post_buckle_mob(mob/buckling_mob)
 	. = ..()
-	owner.add_control_flags(buckling_mob, VEHICLE_CONTROL_DRIVE|VEHICLE_CONTROL_SETTINGS)
 	buckling_mob.reset_perspective(owner)
 	buckling_mob.pixel_x = pixel_x
 	buckling_mob.pixel_y = pixel_y
+	if(vis_range_mod)
+		buckling_mob.client.view_size.set_view_radius_to("[vis_range_mod]x[vis_range_mod]")
```

Comment:
this isnt a modifier it just directly sets it
also iirc isnt there an ACTUAL mod that you could use here?

Reply:
It doesn't directly set, or at least the way it changes your view is not uh, entirely clear. i.e. and value of and 5 result in the same end view area (+2 range). All the viewmod stuff seems really cursed, if there's just a straight forwards way to go '+2 to your view range' I am happy to change it.